frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -144,35 +150,51 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     return null
   }
 
-  static getDeckObject = (props: PropsWithHeight): DeckObject => {
+  static getDeckObject = (
+    props: PropsWithHeight,
+    state: Partial<State>
+  ): DeckObject => {
     const { element, width, height, theme } = props
-    const json = JSON.parse(element.json)
+    const { elementId } = state
+
+    const isFullScreen = Boolean(height)
```

Comment:
While it's currently true that `height` is only set in fullscreen mode, it's probably not the safe way to check if it is in fullscreen. The `withFullScreenWrapper` provides another flag called `isFullScreen`, which is the safer option to use here. E.g. I used this one in the dataframe implementation:

https://github.com/streamlit/streamlit/blob/643970651c11b84f78ff333dc1d3c3217a5b70e5/frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx#L103

Reply:
Thanks will change that.